[Bug]: Folder `tilt` will bock `make dev-env-setup`

### Prerequisites

- [x] I searched existing issues
- [x] I can reproduce this issue

### Bug Description

When I run `make dev-env-setup` in a clean environment, I will meet with an error:
```
ℹ️  Installing Tilt...
Continue? [Y/n] Y
+ curl -fsSL https://github.com/tilt-dev/tilt/releases/download/v0.36.0/tilt.0.36.0.linux.x86_64.tar.gz
+ tar -xzv tilt
tilt
tar: tilt: Cannot open: File exists
tar: Exiting with failure status due to previous errors
make: *** [Makefile:126: dev-env-setup] Error 2
```

I guess folder `tilt` can be removed because `make dev-env-setup` will create this folder.

### Component

Make a dev environment

### Steps to Reproduce

```
git clone https://github.com/nvidia/nvsentinel.git
cd nvsentinel
make dev-env-setup  # Install all dependencies
```

### Environment

- NVSentinel version: main branch
- Kubernetes version: v1.34.2
- Deployment method: make dev-env-setup


### Logs/Output

_No response_

--- a/scripts/setup-dev-env.sh
+++ b/scripts/setup-dev-env.sh
@@ -552,7 +552,12 @@ if [[ "${SKIP_TOOLS}" == "false" ]]; then
             if [[ "${OS}" == "darwin" ]]; then
                 brew install tilt
             elif [[ "${OS}" == "linux" ]]; then
+                # Install to /tmp to avoid conflicts with tilt/ directory in repo
+                TILT_INSTALL_DIR=$(mktemp -d)
+                pushd "${TILT_INSTALL_DIR}" > /dev/null
                 curl -fsSL https://raw.githubusercontent.com/tilt-dev/tilt/master/scripts/install.sh | bash
+                popd > /dev/null
+                rm -rf "${TILT_INSTALL_DIR}"
             fi
             log_success "Tilt installed"
         fi

